GameOfLifeApp/GameLogic.cs

```diff
@@ -0,0 +1,143 @@
+        private int NeighboursCount(int currentRow, int currentColumn)
+        {
+            int count = 0;
+            for (int row = currentRow - 1; row <= currentRow + 1; row++)
+            {
+                for (int column = currentColumn - 1; column <= currentColumn +1; column++)
+                {
+                    int actualRow;
+                    int actualColumn;
+                    if (row < 0)
+                    {
+                        actualRow = currentField.GetLength(0) - 1;
+                    }
+                    else if (row == currentField.GetLength(0))
+                    {
+                        actualRow = 0;
+                    }
+                    else
+                    {
+                        actualRow = row;
+                    }
+                    if (column < 0)
+                    {
+                        actualColumn = currentField.GetLength(1) - 1;
+                    }
+                    else if (column == currentField.GetLength(1))
+                    {
+                        actualColumn = 0;
+                    }
+                    else
+                    {
+                        actualColumn = column;
+                    }
+                    count += currentField[actualRow, actualColumn] ? 1 : 0;
+                }
+            }
+            count -= currentField[currentRow, currentColumn] ? 1 : 0;
+
+            return count;
+        }
+
+        /// <summary>
+        /// Gets new boolean values according to statements.
+        /// </summary>
+        /// <param name="neighboursCount">Number of alive cells around a given cell.</param>
+        /// <param name="isCurrentAlive">Given cell is alive. Value of isCurrentAlive: true.</param>
+        /// <returns>Return the next value for a given cell.</returns>
+        private bool GetCellOffsprings(int neighboursCount, bool isCurrentAlive)
+        {
+            if (!isCurrentAlive && neighboursCount == 3)
+            {
+                return true;
+            }
+            else if (isCurrentAlive && neighboursCount == 2 || neighboursCount == 3)
+            {
+                return true;
+            }
+            return false;
+        }
+
+        /// <summary>
+        /// This method generates a new field.       
+        /// </summary>
+        public void GetNextGeneration()
+        {
+            for (int row = 0; row < countOfRows; row++)
+            {
+                for (int column = 0; column < countOfColumns; column++)
+                {
+                    int count = NeighboursCount(row, column);
+                    bool value = GetCellOffsprings(count, currentField[row, column]);  
+
+                    generatedField[row, column] = value;
+                }
+            }
+        }
+
+        /// <summary>
+        ///  Transfer data to a generated field.       
+        /// </summary>
+        public void TransferNextGenerations()
```

Comment:
Do you ever see this method being used without calling the `GetNextGeneration()` method first or calling the `GetNextGeneration()` without calling this method after? Because, most likely, you can combine these two methods as they typically work in tandem.

Reply:
2 methods are combined.